docs/rackhd/graph_progress.rst

```diff
@@ -0,0 +1,146 @@
+
+.. _Workflow Progress Measurement:
+
+Workflow Progress Measurement
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+RackHD progress information contains two levels of progress as shows in `Progress Message Payload Example`_ :
+
+- `Task level progress`: progress measurement of the executing task of an active workflow.
+- `Workflow level progress`: progress measurement of an active workflow.
+
+Task progress is actually part of a workflow progress. However task and workflow has two independent progress measurement methods.
+
+Workflow level progress measurement
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Workflow level progress is based on tasks counting. It is measured by completed tasks count (which will be assigned to `value`) against total tasks count (which will be assigned to `maximum`) for an active workflow.
+
+Task level progress measurement
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+RackHD provides progress has different task level progress measurement methods for non-long-run tasks and two long-run tasks, OS installation tasks and secure erase task.
+
+**Non-long-run task progress**
+
+Every RackHD has two events:
+
+- `task started`
+- `task finished`
+
+A non-long-run task will complete in short time and only the started and finished events can be sensed. Thus only two progress messages will be published for non-long-run tasks.
+
+Besides task started and finished events, a time-consuming task is not proper to only publish two events, thus different measurements are created.
+
+**OS installation task progress**
+
+As a typical long-run task, OS installation task progress can not be easily measured. As a compromise, RackHD creates some milestones at important timeslot of installation process and divides the task into several sub-tasks.
+
+Take CentOS installation for example, 4 milestones is created for this tasks:
+
+- Profile downloaded: installation task start executing
+- Kernel downloaded: installer kernel downloaded
+- Installation started: installer is started
+- Installation finished: installation and configuration completed
+
+With completion of each subtask, RackHD will send progress information to user with `value` varies from 1 to 4 and `maximum` 4;
+
+**Secure erase task progress**
+
+For secure erase task, RackHD can get continuous progress information with percentage from node. Thus node is required to send the percentage data to RackHD with fixed internal. RackHD will receive and parser the percentage to get `value` and `maximum` and then publish progress message.
+
+Progress Message Retrieving Channels
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As instant data, progress message can't be retrieved via API.
+Instead progress messages will be published in AMQP channel and posted to webhook urls after adding RackHD standard message header.
+
+Below are basic information for user to retrieve data with AMQP
```

Comment:
@pengz1 as a reader with few knowledge in this area,   I give applause to the document as I fully understand the feature after reading it. Minor comments are given, only from the grammar point of view. 

+1 after fixed these minor issues